### PR TITLE
Forbid derived units in target

### PR DESCRIPTION
**Pull request by AlexandrHoroshih**

### CODE REVIEW

File: packages/effector/index.d.ts [hunk 1]
```diff
@@ -155,6 +156,7 @@ export interface Event<Payload> extends Unit<Payload> {
  * The function you can call to trigger an event.
  */
 export interface EventCallable<Payload> extends Event<Payload>, UnitTargetable<Payload> {
+  kind: "event"
```
Comment (Minhir) [inline packages/effector/index.d.ts:159]: Why we don't go futher and introduce `kind: "event-callable"`?

Comment (AlexandrHoroshih) [reply packages/effector/index.d.ts:159]: Why do we need to?

So far it is not really necessary, i think - this move was done only to improve TS type inference in a few edge-cases

Comment (AlexandrHoroshih) [reply packages/effector/index.d.ts:159]: Also there is now a separate runtime field `unit.targetable` 😅

Comment (Minhir) [reply packages/effector/index.d.ts:159]: I thought more about consistency. Never mind, it looks good for me.

**Review by Minhir — APPROVED** [review]

File: src/effector/is.ts [hunk 2]
```diff
@@ -54,9 +54,8 @@ export const assertTarget = (
   targetField: string = 'target',
 ) =>
   forEach(arrifyNodes(target), item =>
-    deprecate(
+    assert(
       !getMeta(item, 'derived'),
-      `${method}: derived unit in "${targetField}"`,
-      `createEvent/createStore`,
+      `${method}: derived unit in "${targetField} is not supported, use createStore/createEvent instead"`,
```
Comment (zerobias) [inline src/effector/is.ts:59]: Quote is missing
```suggestion
      `${method}: derived unit in "${targetField}" is not supported, use createStore/createEvent instead"`,
```

**Pull request merged**